client/my-sites/email/email-providers-stacked-comparison/index.tsx

```diff
@@ -109,16 +107,8 @@ const EmailProvidersStackedComparison = ( {
 			} )
 		);
 
-		page(
-			emailManagementPurchaseNewEmailAccount(
-				selectedSite.slug,
-				selectedDomainName,
-				currentRoute,
-				null,
-				selectedEmailProviderSlug,
-				newIntervalLength
-			)
-		);
+		const queryString = stringify( { interval: newIntervalLength } );
```

Comment:
I think we may be losing the `provider` and `source` parameters in the URL using this approach, and I am not sure if we even support those URL parameters from the email home page... 🤔 
```suggestion
		const queryString = stringify( { interval: newIntervalLength, provider: selectedEmailProviderSlug, source } );
```

I am starting to wonder if all these hoops are necessary, and whether it may be cleaner to manage these values via the stacked comparison component's state.

Reply:
Seems we only support source. We need those parameters for the following cases:

- Source (to know where this component has been called)
- Provider, to expand the provider (AKA when you are coming from in-depth comparison)
- Interval, to default to an interval toggle when accessing the page, also when coming back from in-depth comparison.